### DISCUSSION TITLE
Add an option to encode two bits in a square of 2 * 2 = 4 pixels.

### DISCUSSION THREAD

**Discussion by Nemgathos** [MeViMo/youbit#14] [Ideas]
Because we do not know whether YouTube will one day change its algorithm to compress full HD videos that consist of `1,920 * 1,080 = 2,073,600` pixels, a more robust way to encode files in a video would be to encode one bit of data in a `2 * 2 = 4` pixels square. This would result in `(1,920 / 2) * (1,080 / 2) = 518,400` bits per key frame.

In order to prevent files from becoming about four times at large, I would suggest to use colour coding that does _not_ fall victim to [chroma subsampling](https://en.wikipedia.org/wiki/Chroma_subsampling). Four possible states could be distinguished by using the complementary colours `black (0, 0, 0)` and `white (255, 255, 255)` as well as `red (255, 0, 0)` and `green (0, 255, 0)`.

_Maybe_, the video files can even be compressed _a little bit_ because the probability that information is lost when a full HD video that consists of `2 * 2` pixel squares is uploaded can be _eliminated_.

Would this be a robust way to encode files even if YouTube decides to change its compression algorithm for full HD videos? What do you think?

**Comment by flonle** [comment]
Using 4 pixels squares instead of single pixels could extend YouBit's ability for robustness. However, that is not what it was designed for and I would personally rather see improvements in performance in the other direction (increase information density rather than decreased).
I would still approve a PR that adds this functionality however as it does not hinder anything else.

I am not sure I understand your point on colors. Why would these proposed colors not fall victim to chroma subsampling?
